docker/kafka-setup/build.gradle

```diff
@@ -0,0 +1,42 @@
+plugins {
+    id 'com.palantir.docker'
+    id 'java' // required for versioning
+}
+
+apply from: "../../gradle/versioning/versioning.gradle"
+
+ext {
+    docker_registry = 'linkedin'
+    docker_repo = 'datahub-kafka-setup'
+    docker_dir = 'kafka-setup'
+}
+
+docker {
+    name "${docker_registry}/${docker_repo}:${version}"
+    dockerfile file("${rootProject.projectDir}/docker/${docker_dir}/Dockerfile")
+    files fileTree(rootProject.projectDir) {
+        include "docker/${docker_dir}/*"
+    }
+    tag("-${version}", "${docker_registry}/${docker_repo}:v${version}")
+    tag("-version", "${docker_registry}/${docker_repo}:v${version}")
+
+    // platform('linux/arm64', 'linux/amd64')
+    buildx(true)
+    load(true)
+    push(false)
+}
+tasks.getByPath('docker').dependsOn('build')
```

Comment:
For my understanding -- This basically ensures that the docker build, tag, and publish process happens _only_ once gradle build of the module has been built?

Reply:
Also, where is the 'build' task configured?

Reply:
Its a builtin task from the `java` plugin and in this case does nothing since its just a shell script. This is copy/paste from subprojects that are actually doing work in build. Doesn't harm anything, but I don't believe its necessary however I do know that the versioning code fails if the `java` plugin is missing in general.